client/me/help/help-contact/index.jsx

```diff
@@ -177,6 +183,16 @@ const HelpContact = React.createClass( {
 		this.clearSavedContactForm();
 	},
 
+	prepareDirectlyWidget: function() {
+		if (
+			this.hasDataToDetermineVariation() &&
+			this.getSupportVariation() === SUPPORT_DIRECTLY &&
```

Comment:
Yeah, I've also sort of kicked around the idea of splitting this up so that `<HelpContact>` is a "controller" component that's only responsible for deciding which variation to show and rendering that component.

As an example, it renders `<HelpSupportPlaceholder>` while it gathers info, and ultimately decides to render `<HelpContactDirectlyForm>`. That component unconditionally initializes the Directly widget on mount and continues rendering the same `<HelpSupportPlaceholder>` component (since we don't know if initialization will complete). If initialization succeed it renders the contact form — if it fails it renders an error message. But the parent `<HelpContact>` component will receive a prop change on `isDirectlyFailed` so it could also choose to show the `<HelpContactForumForm>` component at that point.

This may not end up being the best approach, but I'd love to figure out a way to separate "decision-making" logic from "variation-specific" logic. That's the most confusing aspect of this component to me.